.github/workflows/hw_mode_test.yml

```diff
@@ -948,3 +948,63 @@ jobs:
       run: |
         docker exec ${{ env.CONTAINER_NAME }} bash -c "cat /root/occlum/build/test/.fail"
         docker stop ${{ env.CONTAINER_NAME }}
+  
+  Spark_test:
+    if: github.event_name == 'push' || ${{ contains(github.event.pull_request.labels.*.name, 'SGX-hardware-test-required') }}
+    runs-on: ${{ matrix.self_runner }}
+    strategy:
+      matrix:
+        self_runner: [[self-hosted, SGX2-HW]]
+
+    steps:
+    - name: Clean before running
+      run: |
+        sudo chown -R ${{ secrets.CI_ADMIN }} "${{ github.workspace }}"
+
+    - name: Checkout code
+      if: github.event_name == 'push'
+      uses: actions/checkout@v2
+      with:
+        submodules: true
+
+    - name: Checkout code from fork
+      if: ${{ contains(github.event.pull_request.labels.*.name, 'SGX-hardware-test-required') }}
+      uses: actions/checkout@v2
+      with:
+        ref: refs/pull/${{ github.event.pull_request.number }}/merge
+        submodules: true
+    
+    - name: Create container
+      run: |
+        docker pull occlum/occlum:latest-ubuntu20.04
+        if [[ "${{ matrix.self_runner[1] }}" == "SGX1-HW" ]]; then
```

Comment:
Then this "if" looks unnecessary since you only test on SGX 2 machine.

Reply:
Done, I have used "- uses: ./.github/workflows/composite_action/hw"